tests/test_doctests.py

```diff
@@ -30,29 +33,26 @@
 if six.PY3:
     options |= doctest.IGNORE_EXCEPTION_DETAIL
 
-def test_doctests():
-    for filename in filenames:
-        filename = os.path.join(
-            os.path.dirname(os.path.dirname(__file__)),
-            filename)
-        yield do_doctest, filename
 
-def do_doctest(filename):
+@pytest.mark.parametrize('filename', filenames)
+def test_doctests(filename):
+    filename = os.path.join(
+        os.path.dirname(os.path.dirname(__file__)),
+        filename)
     failure, total = doctest.testfile(
         filename, module_relative=False,
         optionflags=options)
     assert not failure, "Failure in %r" % filename
 
-def test_doctest_mods():
-    for module in modules:
-        yield do_doctest_mod, module
 
-def do_doctest_mod(module):
+@pytest.mark.parametrize('module', modules)
+def test_doctest_mods(module):
     module = simple_import(module)
     failure, total = doctest.testmod(
         module, optionflags=options)
     assert not failure, "Failure in %r" % module
 
+
```

Comment:
and two lines between module level methods and other blocks

Reply:
See above.